src/fundamental/macro-fundamental.h

```diff
@@ -376,18 +376,16 @@
 
 /* Takes inspiration from Rust's Option::take() method: reads and returns a pointer, but at the same time
  * resets it to NULL. See: https://doc.rust-lang.org/std/option/enum.Option.html#method.take */
-#define TAKE_GENERIC(var, type, nullvalue)                       \
-        ({                                                       \
-                type *_pvar_ = &(var);                           \
-                type _var_ = *_pvar_;                            \
-                type _nullvalue_ = nullvalue;                    \
-                *_pvar_ = _nullvalue_;                           \
-                _var_;                                           \
+#define TAKE_GENERIC(var, nullvalue)                 \
+        ({                                           \
+                typeof(var) *_pvar_ = &(var);        \
+                typeof(var) _var_ = *_pvar_;         \
+                typeof(var) _nullvalue_ = nullvalue; \
+                *_pvar_ = _nullvalue_;               \
```

Comment:
The extra `_nullvalue_` var is not really useful here. Just assign to pvar directly if you touch this?

Reply:
That's for type safety, I think.

Reply:
Yeah. With automatic typing we'd accept almost anything. So we don't want to do this.